src/ppx/grfx/grfx_swapchain.cpp

```diff
@@ -291,6 +355,25 @@ Result Swapchain::GetRenderPass(uint32_t imageIndex, grfx::AttachmentLoadOp load
     return ppx::SUCCESS;
 }
 
+Result Swapchain::GetRenderTargetView(uint32_t imageIndex, grfx::AttachmentLoadOp loadOp, grfx::RenderTargetView** ppView) const
+{
+    if (!IsIndexInRange(imageIndex, mClearRenderTargets)) {
+        return ppx::ERROR_OUT_OF_RANGE;
+    }
+    if (loadOp == grfx::ATTACHMENT_LOAD_OP_CLEAR) {
+        *ppView = mClearRenderTargets[imageIndex];
+    }
+    else {
+        *ppView = mLoadRenderTargets[imageIndex];
+    }
+    return ppx::SUCCESS;
+}
+
+Result Swapchain::GetDepthStencilView(uint32_t imageIndex, grfx::DepthStencilView** ppView) const
+{
```

Comment:
Should this function range check `imageIndex` as well?

Reply:
Yes, thanks! It also wasn't supposed to get it through the render pass.